tests/esplora.rs

```diff
@@ -33,38 +34,34 @@ async fn test_esplora_wallet() {
     };
 
     let seed = Mnemonic::parse(MNEMONIC).unwrap().to_seed("");
-    let mut wallet = EsploraWallet::from_seed(&seed, NETWORK, ADDRESS_TYPE, esplora_url).expect("esplora_wallet");
+    let mut wallet = Wallet::from_seed(&seed, NETWORK, ADDRESS_TYPE).expect("wallet");
+    let mut blockchain_client = EsploraClient::new(esplora_url).expect("esplora_client");
 
-    wallet.full_scan(STOP_GAP, PARALLEL_REQUESTS).await.expect("full_scan");
+    let block_height = wallet.latest_checkpoint().height();
+    assert_eq!(block_height, 0);
 
-    let address0 = wallet.peek_address(KeychainKind::External, 0);
-    assert_eq!(
-        address0.address(),
-        "tb1qjtgffm20l9vu6a7gacxvpu2ej4kdcsgc26xfdz".to_string()
-    );
-
-    let balance = wallet.balance();
-    assert_eq!(balance.total().to_sat(), 0);
-
-    let address1 = wallet.next_unused_address(KeychainKind::External);
-    assert_eq!(address1.keychain(), KeychainKind::External);
-    assert_eq!(address1.index(), 0);
-
-    let address2 = wallet.reveal_next_address(KeychainKind::External);
-    assert_eq!(address2.index(), 1);
+    let full_scan_request = wallet.start_full_scan();
+    let update = blockchain_client
+        .full_scan(full_scan_request, STOP_GAP, PARALLEL_REQUESTS)
+        .await
+        .expect("full_scan");
+    wallet.apply_update(update).expect("full_scan apply_update_at");
 
-    let address3 = wallet.next_unused_address(KeychainKind::External);
-    assert_eq!(address3.index(), 0);
+    let fullscan_block_height = wallet.latest_checkpoint().height();
+    assert!(fullscan_block_height > 0);
 
-    // Should do a single call to the server (for each keychain)
-    wallet.sync(1).await.expect("sync");
+    wallet.reveal_addresses_to(KeychainKind::External, 5);
 
-    // Should do a stop_gap calls to the server (for each keychain) and not start from beginning
-    wallet
-        .full_scan(STOP_GAP, PARALLEL_REQUESTS)
+    let sync_request = wallet.start_sync_with_revealed_spks();
+    let update = blockchain_client
+        .sync(sync_request, PARALLEL_REQUESTS)
         .await
-        .expect("second full_scan");
+        .expect("sync");
+    wallet
+        .apply_update_at(update, Some((Date::now() / 1000.0) as u64))
```

Comment:
Leaving a suggestion, but we might wanna re-run the linter after this one

```suggestion
        .apply_update(update)
```